pyquil/quilbase.py

```diff
@@ -945,6 +999,14 @@ def out(self) -> str:
     def __str__(self) -> str:
         return super().to_quil_or_debug()
 
+    def __copy__(self) -> Self:
+        return self
+
+    def __deepcopy__(self, memo: Dict) -> "UnaryClassicalInstruction":
+        copy = UnaryClassicalInstruction._from_rs_unary_logic(super().__deepcopy__(memo))
+        copy.__class__ = self.__class__
```

Comment:
Why is this needed in some instructions and not others?

Reply:
`UnaryClassicalInstruction` is a base class for `ClassicalNeg` and others. Without this, `__deepcopy__` on a subclass like `ClassicalNeg` would return a `UnaryClassicalInstruction`. It's not needed on classes that aren't used as a base class.